ckan-backend-dev/src/ckanext-wri/ckanext/wri/logic/action/get.py

```diff
@@ -440,7 +454,18 @@ def notification_get_all(
         )
 
     if not notification_objecst_result:
-        raise logic.NotFound(_("Notification not found"))
+        return []
+    
+    for notification in notification_objecst_result:
```

Comment:
this can actually be considerably faster, if you create a dict for users, datasets, topics, teams you can then add every user/dataset/topic/team to each one of those dicts, and if the user has already been previously fetched you just return the user instead of grabbing from the model, after all calling the database is still kinda of expensive

Reply:
so if there are 50 notifications, and they are all by the same user you would only call the DB once, the code could be something like this

```
users = {}
if users.get(notification["sender_id"]) is None:
    notification_user = model_dictize.user_dictize(model.User.get(notification['sender_id']), context)
    notification['send_obj'] = notification_user
    users[notification['sender_id']] = notification_user
else:
  notification['send_obj'] = users.get(notification["send_id"])
```

Same can be done for datasets/teams/topics